include/preprocess/rounding_util_functions.hpp

```diff
@@ -17,9 +18,34 @@
 #include "Spectra/include/Spectra/MatOp/SparseSymMatProd.h"
 
 
+enum EllipsoidType
+{
+  MAX_ELLIPSOID = 1,
+  LOG_BARRIER = 2,
+  VOLUMETRIC_BARRIER = 3
+};
+
+template <int T>
+struct AssertBarrierFalseType : std::false_type {};
+
 template <typename T>
 struct AssertFalseType : std::false_type {};
 
+template <typename VT, typename NT>
+NT get_max_step(VT const& Ad, VT const& b_Ax)
```

Comment:
this can be written as 
```
template <typename VT>
auto get_max_step(VT const& Ad, VT const& b_Ax)
{
  using NT = VT::Scalar;
  ....
}
```
and then call as `get_max_step(Ad,b_Ax)`

Reply:
Thanks, done